client/my-sites/domains/transfer-domain/index.jsx

```diff
@@ -88,6 +88,9 @@ export class TransferDomain extends Component {
 	};
 
 	componentWillMount() {
+		if ( ! this.props.transferInAllowed ) {
+			page.redirect( '/domains/add/mapping/' + this.props.selectedSiteSlug );
+		}
```

Comment:
Let's also `return null;` in `render()` so we don't get the blink effect, where we render the component, then redirect.